src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -118,42 +128,35 @@ void *frontendservicesesion(void *dummyPt) {
     KafkaConnector *kstream;
     Partitioner graphPartitioner(numberOfPartitions, 1, spt::Algorithms::HASH);
 
-    for (int i = 0; i < workerList.size(); i++) {
-        Utils::worker currentWorker = workerList.at(i);
-        string workerHost = currentWorker.hostname;
-        string workerID = currentWorker.workerID;
-        int workerPort = atoi(string(currentWorker.port).c_str());
-        DataPublisher *workerClient = new DataPublisher(workerPort, workerHost);
-        workerClients.push_back(workerClient);
-    }
+    vector<DataPublisher *> workerClients;
+    bool workerClientsInitialized = false;
+
     bool loop_exit = false;
+    int failCnt = 0;
     while (!loop_exit) {
-        if (currentFESession == Conts::MAX_FE_SESSIONS + 1) {
-            currentFESession--;
-            std::string errorResponse = "Jasminegraph Server is Busy. Please try again later.";
-            int result_wr = write(connFd, errorResponse.c_str(), errorResponse.length());
-            if (result_wr < 0) {
+        if (currentFESession > Conts::MAX_FE_SESSIONS) {
```

Comment:
Shouldn't we take this part of the code outside the `while (!loop_exit){}` ?

Reply:
![image](https://github.com/miyurud/jasminegraph/assets/1388997/62d69ca6-6644-4fdf-ae70-ac6b20139b4a)

```
        if (currentFESession > Conts::MAX_FE_SESSIONS) {
            if (!Utils::send_str_wrapper(connFd, "JasmineGraph server is busy. Please try again later.")) {
                frontend_logger.error("Error writing to socket");
            }
            break;
        }
```

Otherwise this error message will appear in FE sessions which started before reaching the threshold?

Reply:
Done

Reply:
Ok